tests/test_download.py

```diff
@@ -260,27 +259,30 @@ def test_https_connection(self):
     # 3: run with an HTTPS certificate with an unexpected hostname
     # 4: run with an HTTPS certificate that is expired
     # Be sure to offset from the port used in setUp to avoid collision.
-    port1 = str(self.PORT + 1)
-    port2 = str(self.PORT + 2)
-    port3 = str(self.PORT + 3)
-    port4 = str(self.PORT + 4)
-    good_https_server_proc = popen_python(
-        ['simple_https_server.py', port1, good_cert_fname])
-    good2_https_server_proc = popen_python(
-        ['simple_https_server.py', port2, good2_cert_fname])
-    bad_https_server_proc = popen_python(
-        ['simple_https_server.py', port3, bad_cert_fname])
-    expd_https_server_proc = popen_python(
-        ['simple_https_server.py', port4, expired_cert_fname])
-
-    for port in range(self.PORT + 1, self.PORT + 5):
-      utils.wait_for_server('localhost', port)
-
-    relative_target_fpath = os.path.basename(target_filepath)
-    good_https_url = 'https://localhost:' + port1 + '/' + relative_target_fpath
-    good2_https_url = good_https_url.replace(':' + port1, ':' + port2)
-    bad_https_url = good_https_url.replace(':' + port1, ':' + port3)
-    expired_https_url = good_https_url.replace(':' + port1, ':' + port4)
+
+    port1 = self.server_process_handler.port + 1
+    port2 = self.server_process_handler.port + 2
+    port3 = self.server_process_handler.port + 3
+    port4 = self.server_process_handler.port + 4
+
+    good_https_server_handler = utils.TestServerProcess(log=logger,
+        server='simple_https_server.py', port=port1,
+        extra_cmd_args=[good_cert_fname])
+    good2_https_server_handler = utils.TestServerProcess(log=logger,
+        server='simple_https_server.py', port=port2,
+        extra_cmd_args=[good2_cert_fname])
+    bad_https_server_handler = utils.TestServerProcess(log=logger,
+        server='simple_https_server.py', port=port3,
+        extra_cmd_args=[bad_cert_fname])
+    expd_https_server_handler = utils.TestServerProcess(log=logger,
+        server='simple_https_server.py', port=port4,
+        extra_cmd_args=[expired_cert_fname])
+
+    suffix = '/' +  os.path.basename(target_filepath)
+    good_https_url = 'https://localhost:' + str(port1) + suffix
+    good2_https_url = 'https://localhost:' + str(port2) + suffix
+    bad_https_url = 'https://localhost:' + str(port3) + suffix
+    expired_https_url = 'https://localhost:' + str(port3) + suffix
```

Comment:
port number typo -- if this did not come up in tests, maybe check why

Reply:
It is not shown in the tests because we are checking that there is an SSL error and not what exact type is:
```
with self.assertRaises(requests.exceptions.SSLError):
        download.unsafe_download(expired_https_url, target_data_length)
```